### PR TITLE
TCPMappings! and test infrastructure improvements, round 1

### PR DESCRIPTION
**Pull request by kflynn**
`TCPMapping`s work now (including with SNI) and we're combining HTTP KAT backend services into superpods, so instead of starting 80-something of them, we start... 3.

### CODE REVIEW

**Comment by kflynn** [comment]
@richarddli and @nbkrause, if you folks can look at the docs I've added, that'd rock.

File: docs/reference/tcpmappings.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+| Attribute                 | Description               |
+| :------------------------ | :------------------------ |
+| [`weight`](/reference/canary)        | (optional) specifies the (integer) percentage of traffic for this resource that will be routed using this mapping |
+
+The name of the mapping must be unique.
+
+## Example `TCPMapping`s
+
+`TCPMapping` definitions are fairly straightforward. Here's an example that simply relays any connection on port 2222 onward to a Kubernetes service:
+
```
Comment (richarddli) [inline docs/reference/tcpmappings.md]: Is this config also the right config necessary to proxy raw TLS?

File: docs/reference/tcpmappings.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+---
+apiVersion: ambassador/v1
+kind: TCPMapping
+name: quote_mapping
+port: 2222
+tls: some-context
+service: qotm
+```
+
+Note that in this example, Ambassador will contact the `qotm` service using cleartext HTTP, not HTTPS. You can explicitly ask Ambassador to use HTTPS:
```
Comment (richarddli) [inline docs/reference/tcpmappings.md]: in this example above (we should specify if it's the example above or below)

File: docs/reference/tcpmappings.md (hunk 1 above)
Comment (kflynn) [reply docs/reference/tcpmappings.md]: Yup -- if you expect raw TLS you'd probably want to use `service: qotm:443` though.

File: docs/reference/tcpmappings.md (hunk 2 above)
Comment (kflynn) [reply docs/reference/tcpmappings.md]: Will fix -- actually, I need to clean up the docs around ports too.

**Comment by richarddli** [comment]
I added a couple comments on the docs, but the docs LGTM otherwise.

**Pull request merged**